static/js/components/SpectroscopyButtonsForm.jsx

```diff
@@ -0,0 +1,130 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { makeStyles } from "@mui/styles";
+import { useForm } from "react-hook-form";
+import { TextField } from "@mui/material";
+import Button from "./Button";
+import UserPreferencesHeader from "./UserPreferencesHeader";
+import * as profileActions from "../ducks/profile";
+import DeletableChips from "./DeletableChips";
+
+const useStyles = makeStyles(() => ({
+  submitButton: {
+    margin: "1.5rem 0 0 0",
+  },
+  form: {
+    display: "flex",
+    gap: "1rem",
+    flexWrap: "wrap",
+    paddingBottom: "1.5rem",
+  },
+}));
+
+const SpectroscopyButtonsForm = () => {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+  const { spectroscopyButtons } = useSelector(
+    (state) => state.profile.preferences
+  );
+  const {
+    handleSubmit,
+    register,
+    reset,
+
+    formState: { errors },
+  } = useForm();
+
+  const onSubmit = (formValues) => {
+    const currSpectroscopyButtons = spectroscopyButtons || {};
+    currSpectroscopyButtons[formValues.spectroscopyButtonName] = {
+      color: formValues.spectroscopyButtonColor,
+      wavelengths: formValues.spectroscopyButtonWavelengths
+        .split(",")
+        .map(Number),
+    };
+    const prefs = {
+      spectroscopyButtons: currSpectroscopyButtons,
+    };
+    dispatch(profileActions.updateUserPreferences(prefs));
+    reset({
+      spectroscopyButtonName: "",
+    });
+  };
+
+  const onDelete = (buttonName) => {
+    const currSpectroscopyButtons = spectroscopyButtons;
+    delete currSpectroscopyButtons[buttonName];
+    const prefs = {
+      spectroscopyButtons: currSpectroscopyButtons,
+    };
+    dispatch(profileActions.updateUserPreferences(prefs));
+  };
+
+  return (
+    <div>
+      <UserPreferencesHeader
+        title="Spectroscopy Extra Wavelengths"
+        popupText="Select a group of wavelengths, give them a common name and color, and a button will appear on spectroscopy plots for showing those spectral lines on the plot."
+      />
+      <div className={classes.form}>
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <div>
+            <TextField
+              label="Wavelengths"
+              {...register("spectroscopyButtonWavelengths", {
+                required: true,
+              })}
+              name="spectroscopyButtonWavelengths"
+              id="spectroscopyButtonWavelengthInput"
+            />
+            <TextField
```

Comment:
So I would add a little tooltip/help icon to redirect users to the bokeh page that has the list of available colors. As is, hard for people to figure out what to type. If that color list is available in bokeh js somehow, might even want to replace by a dropdown that contains all unused colors?

Reply:
Here is the URL btw: https://docs.bokeh.org/en/latest/docs/reference/colors.html#bokeh-colors-groups

Reply:
If free form, we definitely want some sort of verification upon profile update that the colors used are valid.